bin/workers-preview.js

```diff
@@ -44,6 +45,7 @@ function readJSONFromResponse(res) {
     let data = ''
 
     res.on('data', chunk => {
+      if (res.statusCode >= 400) reject(new Error(data))
```

Comment:
We'll want to move this line into the `res.once('end', () => { ... })` handler. That way, the entire response body will have accumulated before we reject the promise. Also, we'll want to change line to return so that it doesn't continue and try and JSON.parse.

```
if (res.statusCode >= 400) return reject(new Error(data))
```